hnn_core/params.py

```diff
@@ -33,10 +34,50 @@ class Params(dict):
         The parameters json file
     """
 
-    def __init__(self, params_fname):
+    def _read_json_file(self, params_fname):
+        """Read from a .json file (default)"""
         with open(params_fname) as json_data:
             params_input = json.load(json_data)
 
+        return params_input
+
+    def _read_param_file(self, params_fname):
+        """Read from a .param file (legacy)"""
+        params_input = {}
+        with open(params_fname, 'r') as fp:
+            ln = fp.readlines()
+            for l in ln:
+                s = l.lstrip()
+                if s.startswith('#'):
+                    continue
+                sp = s.split(':')
+                if len(sp) > 1:
+                    key = sp[0].strip()
+                    value = sp[1].strip()
+                    if '.' in value or 'e' in value:
+                        try:
+                            params_input[key] = float(value)
+                        except ValueError:
+                            params_input[key] = value
+                    else:
+                        try:
+                            params_input[key] = int(value)
+                        except ValueError:
+                            params_input[key] = value
+
+        return params_input
+
+    def __init__(self, params_fname):
+        try:
+            params_input = self._read_json_file(params_fname)
+        except json.decoder.JSONDecodeError:
+            params_input = self._read_param_file(params_fname)
```

Comment:
sorry I was not clear enough. I was proposing changing a little bit how it works currently. Currently, you construct the param object by doing:

```py
params = Params(params_fname)
```

however, I was proposing to instead do:

```py
from hnn_core import read_params

params_json = read_params('blah.json')
params_legacy = read_params('blah.params')
```

and inside `read_params` you would have:

```py

if ext == '.json': 
    params_dict = _read_json(params_fname)
elif ext == '.params':
   params_dict = _read_legacy_params(params_fname)
else:
   raise ValueError('Unrecognized extension, expected one of .json, .params. Got %s' % ext)
params = Params(params_dict)
```

does that make sense? The advantage of doing this is that you can easily test the params object from Python without always needing a file all the time. It will simplify testing etc.

Reply:
I see. Yes, that makes sense. I actually wanted to do this during the sensitivity analysis for passing a param dictionary between parallel jobs without writing a file.

Reply:
@jasmainak this is coded in updated PR